networkx/algorithms/connectivity/molecular_connectivity.py

```diff
@@ -0,0 +1,53 @@
+"""
+Implements the calculation of a Randic index for a graph._summary_
+"""
+import networkx as nx
+
+from math import sqrt
+from networkx.utils import not_implemented_for
+
+__all__ = ["randic_index"]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def randic_index(G):
+    r"""
+    Compute the Randic index of the graph.
+
+    Parameters
+    ----------
+    G : graph
+        A Networkx graph.
+
+    Returns
+    -------
+    index: int
+        The calculated Randic Index of the graph.
+
+
+    Raises
+    ------
+    NetowrkXNotImplemented
+        The algorithm does not support DiGraph, MultiGraph, and MultiDiGraph
+
+    Examples
+    --------
+    >>> g = nx.complete_graph(5)
+    >>> randic_index(g)
+    2.5
+
+    Notes
+    -----
+    The Randic index [1] is a theoretical characterization of molecular branching, where
+    the molecules can be represented as undirected graphs. The Randic index is calculated as:
+    $R(G) = \sum_{u,v \in E} (deg(u)deg(v))^{-1/2}$.
+
+    References
+    --------
+        [1] Randic, M. (1975). Characterization of molecular branching.
+        Journal of the American Chemical Society, 97(23), 6609-6615.
```

Comment:
Out of curiosity - is this publicly available without a paywall? I've never heard of the Randic index and it'd be good to be able to read the article where it was defined to verify that things are correct here and that the API is sensible.

Reply:
Let's use the DOI.  https://pubs.acs.org/doi/10.1021/ja00856a001
It seems to have no paywall at least for articles that old.

Reply:
That's what I was looking at too... the first page is freely accessible but the article in its entirety is behind a paywall (AFAICT)

Reply:
Yes -- my mistake.  I think it was giving me access via my IP number on campus or something.
You are correct that it is behind a paywall.  Maybe there is another citation we can add to this one that is more recent and explains how it works.

Reply:
Would this article fit the bill?
https://www.tandfonline.com/doi/pdf/10.1016/j.akcej.2017.09.006